app/assets/javascripts/miq_application.js

```diff
@@ -320,7 +320,10 @@ function miqCheckForChanges() {
     }
   }
   // use default browser reaction for onclick
-  return true;
+  if(obj && (href = obj.getAttribute("href")))
```

Comment:
Oh.. that's probably an interaction with the router's HTML5 mode - if you try to go to a route that's not defined in the router, it just redirects you to the default one. I think you could use something like `$rootScope.$on('$stateNotFound', function() { ... })` .. and do a `window.location.href = ...` to go to the intended url if it's not found.